hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go

```diff
@@ -132,7 +133,7 @@ func (p Agent) ReconcileCredentials(ctx context.Context, c client.Client, create
 	role := &rbacv1.Role{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: hcluster.Spec.Platform.Agent.AgentNamespace,
-			Name:      credentialsRBACName,
+			Name:      fmt.Sprintf("%s-%s", CredentialsRBACPrefix, controlPlaneNamespace),
```

Comment:
I see why we need a roleBinding but do we really need a role per controlPlaneNamespace?

Reply:
We don't, different hosted clusters can share the same role, but since we do want to clean it up when deleting the hostedcluster I think it's better if each hostedcluster will have it's own role.